### PR TITLE
Fix running unittests locally

### PR DESCRIPTION
**Pull request by AhmedHanafy725**
### Description

- Add admin marker
- Fix redis client test to get redis config from beside file
- Set admin marker for tests that needs admin

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings

### CODE REVIEW

**Review by waleedhammam — APPROVED** [review]

**Comment by abom** [comment]
Still fails locally:

```
make unittests
```

Output:

```
....
============================================================== short test summary info ==============================================================
FAILED tests/core/executors/test_tmux.py::TestTmux::test005_execute_in_window - AssertionError: False is not true
FAILED tests/sals/process/test_process.py::ProcessTests::test_07_check_running - AssertionError: True is not false
FAILED tests/sals/process/test_process.py::ProcessTests::test_09_get_process_environ - AssertionError: 2 != 1 : tail didn't start
FAILED tests/sals/process/test_process.py::ProcessTests::test_10_get_filtered_pids - AssertionError: False is not true
FAILED tests/sals/process/test_process.py::ProcessTests::test_13_get_kill_process_by_port - AssertionError: False is not true
FAILED tests/sals/process/test_process.py::ProcessTests::test_18_ps_find - AssertionError: 2 != 1
FAILED tests/sals/process/test_process.py::ProcessTests::test_20_get_ports_mapping - AssertionError: False is not true
================================== 7 failed, 186 passed, 3 skipped, 14 deselected, 9 warnings in 91.57s (0:01:31) ===================================
Makefile:10: recipe for target 'unittests' failed
make: *** [unittests] Error 1
```

File: tests/clients/redis/test_redis.py [hunk 1]
```diff
@@ -10,29 +10,28 @@
 
 
 class RedisTests(BaseTests):
+    def setUp(self):
+        self.redis_instance_name = self.randstr()
+        self.redis_config_target_path = "/tmp/redis.conf"
+        self.port = randint(1000, 2000)
```
Comment (abom) [inline tests/clients/redis/test_redis.py:16]: IMHO, this range should'not include ports < 1024, also, it should not be random, might cause problems on different systems, let's choose a port which is not well known and used by any other possible server.

**Comment by waleedhammam** [comment]
> Still fails locally:
> 
> ```
> make unittests
> ```
> 
> Output:
> 
> ```
> ....
> ============================================================== short test summary info ==============================================================
> FAILED tests/core/executors/test_tmux.py::TestTmux::test005_execute_in_window - AssertionError: False is not true
> FAILED tests/sals/process/test_process.py::ProcessTests::test_07_check_running - AssertionError: True is not false
> FAILED tests/sals/process/test_process.py::ProcessTests::test_09_get_process_environ - AssertionError: 2 != 1 : tail didn't start
> FAILED tests/sals/process/test_process.py::ProcessTests::test_10_get_filtered_pids - AssertionError: False is not true
> FAILED tests/sals/process/test_process.py::ProcessTests::test_13_get_kill_process_by_port - AssertionError: False is not true
> FAILED tests/sals/process/test_process.py::ProcessTests::test_18_ps_find - AssertionError: 2 != 1
> FAILED tests/sals/process/test_process.py::ProcessTests::test_20_get_ports_mapping - AssertionError: False is not true
> ================================== 7 failed, 186 passed, 3 skipped, 14 deselected, 9 warnings in 91.57s (0:01:31) ===================================
> Makefile:10: recipe for target 'unittests' failed
> make: *** [unittests] Error 1
> ```

Screenshot from 2020-10-13 22-48-09

weird, sure you pulled latest `development_fixRunningTestsLocally` ?

**Comment by abom** [comment]
Don't know if running this tests inside a tmux would make any difference, also, @waleedhammam what's your current python version?

**Comment by waleedhammam** [comment]
> Don't know if running this tests inside a tmux would make any difference, also, @waleedhammam what's your current python version?

it's 3.8.2 ubuntu 20 outside tmux, it think yes it'll be different, cause there are tmux tests in there

**Comment by abom** [comment]
Outside tmux, python3.6

```
============================================================== short test summary info ==============================================================
FAILED tests/sals/process/test_process.py::ProcessTests::test_07_check_running - AssertionError: True is not false
FAILED tests/sals/process/test_process.py::ProcessTests::test_09_get_process_environ - AssertionError: 2 != 1 : tail didn't start
FAILED tests/sals/process/test_process.py::ProcessTests::test_10_get_filtered_pids - AssertionError: False is not true
FAILED tests/sals/process/test_process.py::ProcessTests::test_13_get_kill_process_by_port - AssertionError: False is not true
FAILED tests/sals/process/test_process.py::ProcessTests::test_15_get_kill_process_by_pids - AssertionError: 2 != 1
FAILED tests/sals/process/test_process.py::ProcessTests::test_20_get_ports_mapping - AssertionError: False is not true
================================== 6 failed, 187 passed, 3 skipped, 14 deselected, 9 warnings in 111.02s (0:01:51) ==================================
Makefile:10: recipe for target 'unittests' failed
make: *** [unittests] Error 1
```

**Comment by waleedhammam** [comment]
> Outside tmux, python3.6
> 
> ```
> ============================================================== short test summary info ==============================================================
> FAILED tests/sals/process/test_process.py::ProcessTests::test_07_check_running - AssertionError: True is not false
> FAILED tests/sals/process/test_process.py::ProcessTests::test_09_get_process_environ - AssertionError: 2 != 1 : tail didn't start
> FAILED tests/sals/process/test_process.py::ProcessTests::test_10_get_filtered_pids - AssertionError: False is not true
> FAILED tests/sals/process/test_process.py::ProcessTests::test_13_get_kill_process_by_port - AssertionError: False is not true
> FAILED tests/sals/process/test_process.py::ProcessTests::test_15_get_kill_process_by_pids - AssertionError: 2 != 1
> FAILED tests/sals/process/test_process.py::ProcessTests::test_20_get_ports_mapping - AssertionError: False is not true
> ================================== 6 failed, 187 passed, 3 skipped, 14 deselected, 9 warnings in 111.02s (0:01:51) ==================================
> Makefile:10: recipe for target 'unittests' failed
> make: *** [unittests] Error 1
> ```

the tmux one succeeded, now we may have the process isn't up yet, @AhmedHanafy725 gonna check about the speed/time limitations

File: tests/clients/redis/test_redis.py (hunk 1 above)
Comment (sameh-farouk) [reply tests/clients/redis/test_redis.py:16]: I would suggest using a more reliable way to get a free port.
currently, I'm using this code in my unit tests for nettools
# getting a random free port on localhost
```
with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
        s.bind(("127.0.0.1", 0))
        port = s.getsockname()[1]
```
if you want a free port to bind to the local machine ipv6 address ()
```
with socket.socket(socket.AF_INET6, socket.SOCK_STREAM) as s:
        s.bind(("::1", 0))
        port = s.getsockname()[1]
```
we need to make it a separate function, maybe in nettools ?

Comment (sameh-farouk) [reply tests/clients/redis/test_redis.py:16]: > > Outside tmux, python3.6
> > ```
> > ============================================================== short test summary info ==============================================================
> > FAILED tests/sals/process/test_process.py::ProcessTests::test_07_check_running - AssertionError: True is not false
> > FAILED tests/sals/process/test_process.py::ProcessTests::test_09_get_process_environ - AssertionError: 2 != 1 : tail didn't start
> > FAILED tests/sals/process/test_process.py::ProcessTests::test_10_get_filtered_pids - AssertionError: False is not true
> > FAILED tests/sals/process/test_process.py::ProcessTests::test_13_get_kill_process_by_port - AssertionError: False is not true
> > FAILED tests/sals/process/test_process.py::ProcessTests::test_15_get_kill_process_by_pids - AssertionError: 2 != 1
> > FAILED tests/sals/process/test_process.py::ProcessTests::test_20_get_ports_mapping - AssertionError: False is not true
> > ================================== 6 failed, 187 passed, 3 skipped, 14 deselected, 9 warnings in 111.02s (0:01:51) ==================================
> > Makefile:10: recipe for target 'unittests' failed
> > make: *** [unittests] Error 1
> > ```
> 
> the tmux one succeeded :D, now we may have the process isn't up yet, @AhmedHanafy725 gonna check about the speed/time limitations

sleep for a sec before checking the process? increase the wait time would be helpful too.

**Comment by sameh-farouk** [comment]
> > Outside tmux, python3.6
> > ```
> > ============================================================== short test summary info ==============================================================
> > FAILED tests/sals/process/test_process.py::ProcessTests::test_07_check_running - AssertionError: True is not false
> > FAILED tests/sals/process/test_process.py::ProcessTests::test_09_get_process_environ - AssertionError: 2 != 1 : tail didn't start
> > FAILED tests/sals/process/test_process.py::ProcessTests::test_10_get_filtered_pids - AssertionError: False is not true
> > FAILED tests/sals/process/test_process.py::ProcessTests::test_13_get_kill_process_by_port - AssertionError: False is not true
> > FAILED tests/sals/process/test_process.py::ProcessTests::test_15_get_kill_process_by_pids - AssertionError: 2 != 1
> > FAILED tests/sals/process/test_process.py::ProcessTests::test_20_get_ports_mapping - AssertionError: False is not true
> > ================================== 6 failed, 187 passed, 3 skipped, 14 deselected, 9 warnings in 111.02s (0:01:51) ==================================
> > Makefile:10: recipe for target 'unittests' failed
> > make: *** [unittests] Error 1
> > ```
> 
> the tmux one succeeded :D, now we may have the process isn't up yet, @AhmedHanafy725 gonna check about the speed/time limitations

you could sleep for a sec before checking the process. increase the wait time would be helpful too.

File: tests/clients/redis/test_redis.py (hunk 1 above)
Comment (abom) [reply tests/clients/redis/test_redis.py]: @sameh-farouk  You can add this as a helper function in nettools to be used later.

**Review by abom — APPROVED** [review]

**Pull request merged**